ezyrb/database.py

```diff
@@ -14,66 +17,35 @@ class Database():
         None meaning no scaling.
     :param array_like space: the input spatial data
     """
-    def __init__(self,
-                 parameters=None,
-                 snapshots=None,
-                 scaler_parameters=None,
-                 scaler_snapshots=None,
-                 space=None):
-        self._parameters = None
-        self._snapshots = None
-        self._space = None
-        self.scaler_parameters = scaler_parameters
-        self.scaler_snapshots = scaler_snapshots
-
-        # if only parameters or snapshots are provided
-        if (parameters is None) ^ (snapshots is None):
-            raise RuntimeError(
-                'Parameters and Snapshots are not both provided')
-
-        if space is not None and snapshots is None:
-            raise RuntimeError(
-                'Snapshot data is not provided with Spatial data')
-
-        if parameters is not None and snapshots is not None:
-            if space is not None:
-                self.add(parameters, snapshots, space)
-            else:
-                self.add(parameters, snapshots)
+    def __init__(self, parameters=None, snapshots=None):
+        self._pairs = []
 
+        if parameters is None and snapshots is None:
+            return
 
-    @property
-    def parameters(self):
-        """
-        The matrix containing the input parameters (by row).
-
-        :rtype: numpy.ndarray
-        """
-        if self.scaler_parameters:
-            return self.scaler_parameters.fit_transform(self._parameters)
+        if len(parameters) != len(snapshots):
+            raise ValueError
 
-        return self._parameters
+        for param, snap in zip(parameters, snapshots):
+            self.add(Parameter(param), Snapshot(snap))
 
     @property
-    def snapshots(self):
+    def parameters_matrix(self):
         """
-        The matrix containing the snapshots (by row).
+        The matrix containing the input parameters (by row).
 
         :rtype: numpy.ndarray
         """
-        if self.scaler_snapshots:
-            return self.scaler_snapshots.fit_transform(self._snapshots)
-
-        return self._snapshots
+        return np.asarray([pair[0].values for pair in self._pairs])
 
     @property
-    def space(self):
+    def snapshots_matrix(self):
```

Comment:
typo, could be changed to `snapshot_matrix`

Reply:
I would keep the plural form at the moment, just to avoid changing all the tests! I'll refactor this class a bit.